idb/postgres/internal/writer/writer.go

```diff
@@ -303,19 +303,19 @@ func writeAccountDeltas(round basics.Round, accountDeltas *ledgercore.AccountDel
 
 }
 
-func writeBoxMods(kvMods map[string]*string, batch *pgx.Batch) error {
+func writeBoxMods(kvMods map[string]ledgercore.ValueDelta, batch *pgx.Batch) error {
 	// INSERT INTO / UPDATE / DELETE FROM `app_box`
 	// WARNING: kvMods can in theory support more general storage types than app boxes.
 	// However, here we assume that all the provided kvMods represent app boxes.
 	// If a non-box is encountered inside kvMods, an error will be returned and
 	// AddBlock() will fail with the import getting stuck at the corresponding round.
-	for key, value := range kvMods {
+	for key, valueDelta := range kvMods {
 		app, name, err := logic.SplitBoxKey(key)
 		if err != nil {
 			return fmt.Errorf("writeBoxMods() err: %w", err)
 		}
-		if value != nil {
-			batch.Queue(upsertAppBoxStmtName, app, []byte(name), []byte(*value))
+		if valueDelta.Data != nil {
```

Comment:
NOTE: the indexer box update code _DOES NOT_  populate `ValueDelta.OldData`. I believe that this isn't necessary as rows are deleted from the `app_box` and no information is retained about old box values.

Reply:
@tzaffi As a point of clarification and to test my understanding:  I agree indexer does _not_ populate `OldData`, but I don't understand the rationale that follows.

Testing my understanding - Does the following make sense to you?

I expect indexer to be provided `OldData` populated by go-algorand indexer.  Which is to say, I don't think indexer _ever_ populates `OldData`.